roles/build_push_container/defaults/main.yml

```diff
@@ -0,0 +1,48 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# All variables intended for modification should be placed in this file.
+# All variables within this role should have a prefix of "cifmw_build_push_container"
+
+# Meta vars
+cifmw_build_push_container_basedir: "{{ cifmw_basedir | default(ansible_user_dir ~ '/ci-framework-data') }}"
+cifmw_build_push_container_artifacts: "{{ (cifmw_build_push_container_basedir, 'artifacts', 'build_push_container')| ansible.builtin.path_join }}"
+
+# Build vars
+cifmw_build_push_container_name:
+cifmw_build_push_container_build_context_path: "{{ (ansible_user_dir, zuul.project.src_dir) | ansible.builtin.path_join }}"
+cifmw_build_push_container_containerfile_name: Containerfile
+cifmw_build_push_container_containerfile_path: >-
+  {{
+    (ansible_user_dir,
+    zuul.project.src_dir,
+    cifmw_build_push_container_containerfile_name
+    ) | ansible.builtin.path_join
+  }}
+cifmw_build_push_container_local_build_tag: "{{ (cifmw_build_push_container_name,'latest') | join(':') }}"
+cifmw_build_push_container_supported_platform: [linux/amd64]
+cifmw_build_push_container_qemu_user_static_image: quay.rdoproject.org/ci-framework/qemu-user-static:latest
+
+# Tagging vars
+cifmw_build_push_container_git_sha: "{{ zuul.commit_id }}"
+cifmw_build_push_container_patch_number: "{{ zuul.change }}"
+cifmw_build_push_container_tag_override: []
+# Push vars
+cifmw_build_push_container_push: false
+cifmw_build_push_container_registry_name:
+cifmw_build_push_container_registry_username:
+cifmw_build_push_container_registry_password:
```

Comment:
i'd declare these as `null`, `None` or `!!null` instead of having them blank

Reply:
I initially had null but `ansible-lint` actually doesn't want it and removed it when running `ansible-lint --fix`

Reply:
I looked into this some more, I'm happy to change it but his bare default variable is being enforced by `ansible-lint --fix` when it calls `yaml-lint` with it's special custom config that is compatible with ansible/Jinja1/yaml [2]

Many of our roles use "" as well, if we want to use `null` for example we should move all roles and add a custom yamllint config.

[1] https://ansible.readthedocs.io/projects/lint/rules/yaml/
[2] https://ansible.readthedocs.io/projects/lint/rules/yaml/#yamllint-configuration